QuestionAnswering/QA_SEQ2SEQ/data_generation/locate_then_ask/ontobuiltenv/synthetic/property_usage.py

```diff
@@ -0,0 +1,29 @@
+from decimal import Decimal
+import random
+from typing import Tuple
+
+import numpy as np
+from constants.namespaces import OBE
+from constants.ontobuiltenv import OBE_PROPERTYUSAGE_LABELS
+from locate_then_ask.ontobuiltenv.model import OBEPropertyUsage
+
+
+class OBEPropertyUsageSynthesizer:
+    PROPERTY_USAGE_CONCEPTS = [OBE + x for x in OBE_PROPERTYUSAGE_LABELS]
+
+    def make(self) -> Tuple[OBEPropertyUsage, ...]:
+        n = random.randrange(1, 4)
+        uses = random.sample(self.PROPERTY_USAGE_CONCEPTS, k=n)
+        shares = [random.randint(1, 100) for _ in range(n)]
+        shares = np.array(shares) / sum(shares)
+
+        assert len(uses) == len(set(uses))
```

Comment:
See above comment.

Reply:
I've replaced the `assert` statement with `raise ValueError(...)`.